python-api/tests/integrationtests/baseclass.py

```diff
@@ -0,0 +1,412 @@
+
+        self.log.info("comparison failed")
+        return (False, img, diff)
+
+    def expect_caps(self, port, expected):
+        """Fetch a sample-frame from the server and return its
+        caps as string"""
+        self.log.debug("building caps-expecting commandline")
+
+        command = ("gst-launch-1.0 "
+                   "tcpclientsrc host=localhost port=%u ! "
+                   "gdpdepay ! "
+                   "%s ! "
+                   "fakesink num-buffers=1") % (port, expected)
+
+        self.log.info("running command for up to 5s")
+        self.log.debug("%s", command)
+        proc = subprocess.Popen(shlex.split(command))
+        try:
+            proc.wait(timeout=5)
+        except subprocess.TimeoutExpired:
+            proc.kill()
+            proc.wait()
+
+        returncode = proc.poll()
+        self.log.debug("command finished, returncode=%d", returncode)
+
+        assert returncode == 0, ("caps received from port %u don't meet "
+                                 "expectation: %s" % (port, expected))
+
+    def is_running_in_ci(self):
+        """Test if the testsuite is ran by Travis-CI"""
+        return os.environ.get('CI', "false") == "true"
+
+    def save_images(self, basename, img, expected, diff):
+        """Save and (in Travis) upload Failed images"""
+
+        basename = os.path.splitext(os.path.basename(basename))[0]
+
+        filename = '%s-a_img.png' % basename
+        self.log.debug("saving to %s", filename)
+        scipy.misc.imsave(
+            filename,
+            img.astype(numpy.uint8))
+
+        if self.is_running_in_ci():
+            self.log.warn("uploaded failed frame")
+            self.upload_image(filename)
+        else:
+            self.log.warn("saved failed frame as: %s",
+                          filename)
+
+        filename = '%s-b_expected.png' % (os.path.splitext(basename)[0])
+        self.log.debug("saving to %s", filename)
+        scipy.misc.imsave(
+            filename,
+            expected.astype(numpy.uint8))
+
+        if self.is_running_in_ci():
+            self.log.warn("uploaded expected frame")
+            self.upload_image(filename)
+        else:
+            self.log.warn("saved expected frame as: %s",
+                          filename)
+
+        filename = '%s-c_diff.png' % (os.path.splitext(basename)[0])
+        self.log.debug("saving to %s", filename)
+        scipy.misc.imsave(
+            filename,
+            diff.astype(numpy.uint8))
+
+        if self.is_running_in_ci():
+            self.log.warn("uploaded difference frame")
+            self.upload_image(filename)
+        else:
+            self.log.warn("saved difference frame as: %s",
+                          filename)
+
+    def upload_image(self, filename):
+        """ upload the specified image to imgurl"""
+        self.log.debug("calling imgurbash.sh %s", filename)
+        subprocess.call([
+            'bash',
+            './imgurbash.sh',
```

Comment:
Yeah, we are going to have TODOs in our code - disabling that warning globally is reasonable.